controllers/propagator/propagation.go

```diff
@@ -122,22 +159,53 @@ func (r *PolicyReconciler) cleanUpPolicy(instance *policiesv1.Policy) error {
 		return err
 	}
 
+	policiesChan := make(chan policiesv1.Policy, len(replicatedPlcList.Items))
+	deletionResultsChan := make(chan deletionResult, len(replicatedPlcList.Items))
+
+	numWorkers := common.GetNumWorkers(len(replicatedPlcList.Items), concurrencyPerPolicy)
```

Comment:
Could you please move this code to just above `for i := 0; i < numWorkers; i++ {`? It's easier to read when a variable is declared where it is used.